pants

```diff
@@ -45,12 +44,41 @@ function tempdir {
   mktemp -d "$1"/pants.XXXXXX
 }
 
+function get_pants_ini_config_value {
+  config_key="$1"
+  # When a line with `config_key: ` is found, replace all the text from the start up to
+  # and including the `: ` with an empty string, then print.
+  sed -ne "/\s*${config_key}: / s#^.*: ##p" pants.ini
+}
+
+function get_python_major_minor_version {
+  python_exe="$1"
+  "$python_exe" <<EOF
+import sys
+major_minor_version = ''.join(str(version_num) for version_num in sys.version_info[0:2])
+print(major_minor_version)
+EOF
+}
+
+function determine_pants_runtime_python_version {
+  interpreter_version="$(get_pants_ini_config_value pants_runtime_python_version)"
+  if [[ -z "${interpreter_version}" ]]; then
+    interpreter_version='2.7'
+  fi
+  python_bin_name=${PYTHON:-python${interpreter_version}}
+  if which "${python_bin_name}" >/dev/null; then
+    echo `which ${python_bin_name}`
```

Comment:
Backtick substitution is "deprecated": http://mywiki.wooledge.org/BashFAQ/082. Many people cargo cult backticking from other scripts, especially for `which` calls, so you'll still see this often, but it should be converted to `$()`.